scripts/build-js/heroicons-js-base.js

```diff
@@ -0,0 +1,43 @@
+const heroiconsLoader = {
+  replaceAllIcons: () => {
+    const getAttrs = (element) =>
+    Array.from(element.attributes).reduce((attrs, attr) => {
+      attrs[attr.name] = attr.value;
+      return attrs;
+    }, {});
+
+    const replaceElement = (element) => {
+      const attributes = getAttrs(element);
+      const iconName = attributes["data-heroicons"];
+
+      const classNames =
+        element.classList.length > 0
+          ? element.classList.value.split(" ")
+          : null;
+
+      const isOutline =
+        attributes["data-type"] && attributes["data-type"] === "outline";
```

Comment:
The attribute name `data-type` seems very generic to me and may conflict with other libraries

Reply:
Have updated to `icon-type`. :)